packages/react/src/use-chat.ui.test.tsx

```diff
@@ -1888,3 +1888,83 @@ describe('test sending additional fields during message submission', () => {
     ),
   );
 });
+
+describe('initialMessages stability', () => {
+  let renderCount = 0;
+
+  const TestComponent = () => {
+    renderCount++;
+    const [derivedState, setDerivedState] = useState<string[]>([]);
+
+    const { messages } = useChat({
+      api: '/api/chat',
+      id: 'test-stability',
+      initialMessages: [
+        {
+          id: 'test-msg-1',
+          content: 'Test message',
+          role: 'user',
+        },
+        {
+          id: 'test-msg-2',
+          content: 'Test response',
+          role: 'assistant',
+        },
+      ],
+    });
+
+    useEffect(() => {
+      setDerivedState(messages.map(m => m.content));
+    }, [messages]);
+
+    if (renderCount > 10) {
+      throw new Error('Excessive renders detected; likely an infinite loop!');
+    }
```

Comment:
needed? or could we just let the test time out?

Reply:
this was a result of the test not timing out as expected! when we `render` the component, it just hangs (due to the infinite render) - when we detect infinite loop this `if` conditional forces the test to fail!